compliance/disallowed_images/disallowed_cloud_images.pt

```diff
@@ -0,0 +1,255 @@
+          locked = 'Yes';
+        }else{
+          locked = 'No';
+        }
+         var instance_image_href = instance.image_href
+	     var matched_image = _.find(ds_images_list, function(image) {return image.href == instance_image_href})
+	    if (matched_image != null && matched_image !== undefined) {
+	     var resource_uid = matched_image.resource_uid
+         var is_cloud_image_resource_uid_matched=false
+        if((param_allowed_cloud_images.indexOf(resource_uid) !== -1) || (param_allowed_cloud_images.join()).match(resource_uid)){
+         is_cloud_image_resource_uid_matched = true;
+        }
+        }
+        if(matched_image != null && !(is_cloud_image_resource_uid_matched)){
+            content.push({
+             cloud_name:instance['cloud_name']
+             id:instance['id']
+             cloud_type:instance['cloud_type']
+             name:instance['name']
+             tags:(dis_tags.slice(2))
+             href:instance['href']
+             locked:locked
+             state:instance['state']
+            })   
+	    }
+	    }
+	}
+  EOS
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "esc_report_disallowed_cloud_images" do
+  email $param_email
+end
+
+escalation "esc_approve_terminate_instances" do
+  request_approval  do
+  label "Approve Instance Termination"
+    description "Approve escalation to run RightScale Cloud Workflow to terminate Instances under Disallowed Cloud Images"
+    parameter "approval_reason" do
+      type "string"
+      label "Reason for Approval"
+      description "[Note]: Locked instances will not be terminated."
+    end
+  end
+  run "terminate_instance", data
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_list_cloud_instances" do
+  validate $ds_list_cloud_instances do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Disallowed Cloud Images Instance Details"
+    detail_template <<-EOS
+# Instance Details
+| Cloud Name | ID | Cloud Type | State | Instance Name | Is Instance Locked | Href | Tags |
+| ---------- | -- | ---------- | ----- | ------------- | ------------------ | ---- | ---- |
+{{ range data -}}
+| {{.cloud_name}} | {{.id}} | {{.cloud_type}} | {{.state}} | {{.name}} | {{.locked}} | {{.href}} | {{.tags}} |
+{{ end -}}
+    EOS
+    escalate $esc_report_disallowed_cloud_images
+    escalate $esc_approve_terminate_instances
+    check eq(size(data),0)
+  end
+end
+
+###############################################################################
+# Cloud Workflow
+###############################################################################
+
+define terminate_instance($data) do
+  foreach $item in $data do
+    sub on_error: skip do
```

Comment:
I believe an error will occur if you try to terminate a locked instance. You may need to skip locked instances.

Reply:
Taken care for locked instance